app/providers/atmosphere/optimization_strategy/default.rb

```diff
@@ -105,7 +120,7 @@ def select
           best_tenant = nil
           best_flavor = nil
           instantiation_cost = Float::INFINITY
-          tmpls.each do |tmpl|
+          @tmpls.each do |tmpl|
```

Comment:
You seem to be confusing `@tmpls` with `@tmpl`. Please re-read your original comment and make sure it doesn't contain any typos in variable names. :)

If you want me to remove `attr_reader :tmpls` then I'll do so. I don't think it's needed anymore.